overlord/registrystate/transaction.go

```diff
@@ -0,0 +1,206 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+/*
+ * Copyright (C) 2023-2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package registrystate
+
+import (
+	"sync"
+
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/registry"
+)
+
+// Transaction performs read and writes to a databag in an atomic way.
+type Transaction struct {
+	Pristine registry.JSONDataBag `json:"pristine,omitempty"`
+
+	RegistryAccount string `json:"registry-account,omitempty"`
+	RegistryName    string `json:"registry-name,omitempty"`
+
+	Modified      registry.JSONDataBag     `json:"modified,omitempty"`
+	Deltas        []map[string]interface{} `json:"deltas,omitempty"`
+	AppliedDeltas int                      `json:"applied-deltas,omitempty"`
+
+	mu sync.RWMutex
+}
```

Comment:
Instead of exporting fields, could we instead do something similar to `type marshalledChange struct` so we don't export fields unnecessarily.

Reply:
I've changed this to hide most of the fields. Thanks for the suggestion